src/overlays/actors/ovl_Obj_Comb/z_obj_comb.c

```diff
@@ -32,71 +34,538 @@ const ActorInit Obj_Comb_InitVars = {
+
+        if (Rand_ZeroOne() < 0.2f) {
+            phi_v0 = 0x40;
+        } else {
+            phi_v0 = 0x20;
+        }
+
+        EffectSsKakera_Spawn(globalCtx, &spC8, &spBC, &spC8, phi_s0, phi_v0, 40, 0, 0, temp_f10, 0, 0, 80, -1,
+                             OBJECT_COMB, object_comb_DL_001040);
+    }
+}
+
+void func_8098D6E0(ObjComb* this, GlobalContext* globalCtx) {
+    s32 i;
+    Vec3f sp70;
+    f32 temp_f0;
+    f32 temp_f20;
+    s32 temp;
+
+    sp70.y = this->actor.world.pos.y + this->actor.depthInWater;
+
+    for (temp = 0, i = 0; i < 8; i++, temp += 0x2000) {
+        temp_f0 = Rand_ZeroOne();
+        temp_f20 = ((1.0f - SQ(temp_f0)) * 14.0f) + 4.0f;
+        sp70.x = (Math_SinS((s32)(Rand_ZeroOne() * 8000.0f) + temp) * temp_f20) + this->actor.world.pos.x;
+        sp70.z = (Math_CosS((s32)(Rand_ZeroOne() * 8000.0f) + temp) * temp_f20) + this->actor.world.pos.z;
+        EffectSsGSplash_Spawn(globalCtx, &sp70, NULL, NULL, 0, 200);
+    }
+
+    sp70.x = this->actor.world.pos.x;
+    sp70.z = this->actor.world.pos.z;
+    EffectSsGSplash_Spawn(globalCtx, &sp70, NULL, NULL, 0, 300);
+}
+
+void func_8098D870(ObjComb* this, GlobalContext* globalCtx) {
+    s32 temp_v0 = func_800A8150(OBJCOMB_GET_3F(&this->actor));
+
+    if (temp_v0 >= 0) {
+        Item_DropCollectible(globalCtx, &this->actor.world.pos, ((OBJCOMB_GET_7F00(&this->actor)) << 8) | temp_v0);
+    }
+}
+
+void func_8098D8C8(ObjComb* this, GlobalContext* globalCtx) {
+    s32 params = OBJCOMB_GET_F(&this->actor);
+    s32 i;
+    s32 phi_s5;
+
+    if (this->actionFunc == func_8098DC60) {
+        phi_s5 = 2;
+    } else {
+        phi_s5 = 1;
+    }
+    this->unk_1B6 = 0;
+
+    for (i = 0; i < params; i++) {
+        if (Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_BEE, this->actor.world.pos.x, this->actor.world.pos.y,
+                        this->actor.world.pos.z, 0, 0, 0, phi_s5)) {
+            this->unk_1B6 = 1;
+        }
+    }
+}
+
+void func_8098D99C(ObjComb* this, GlobalContext* globalCtx) {
+    s32 params;
+    Actor* temp_v0;
+
+    if (func_8098CE40(this, globalCtx)) {
+        params = (OBJCOMB_GET_1F(&this->actor) << 2) | 0xFF01;
+        temp_v0 = Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_SW, this->actor.world.pos.x,
+                              this->actor.world.pos.y, this->actor.world.pos.z, 0, this->actor.home.rot.y, 0, params);
+
+        if (temp_v0 != NULL) {
+            temp_v0->parent = &this->actor;
+            if (this->actionFunc == func_8098DC60) {
+                temp_v0->velocity.y = 8.0f;
+                temp_v0->speedXZ = 2.0f;
+            } else {
+                temp_v0->velocity.y = 10.0f;
+                temp_v0->speedXZ = 2.0f;
+            }
+            this->unk_1B6 = 1;
+            play_sound(NA_SE_SY_TRE_BOX_APPEAR);
+        }
+    }
+}
+
+void func_8098DA74(ObjComb* this, GlobalContext* globalCtx) {
+    s32 temp_v0 = OBJCOMB_GET_8000(&this->actor) | OBJCOMB_GET_80(&this->actor);
```

Comment:
```
    temp_v1 = (((s32) temp_v0 >> 0xE) & 2) | (((s32) temp_v0 >> 7) & 1);
```
Weird... if you shift params by 0xE you should be left with only the farthest left bit which is 0x8000 but its bitwise anding by two? Is this checking if a signed params is negative?